### PR TITLE
Create projects map page layout and route

### PR DESCRIPTION
**Pull request by mddilley**
## This is the collector branch for the projects map MVP

This will not be merged until the MVP is complete and ready for release.

### Merge TODOs
- [x] Add to QA test script
- [x] Merge main and test migrations (update if needed)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9913
https://github.com/cityofaustin/atd-data-tech/issues/9914
https://github.com/cityofaustin/atd-data-tech/issues/9919

This adds: 
- a **Map** tab to the main view that links to a new route
- a map to start building on
- a collapsible side bar that has a placeholder title and content

## Testing
**URL to test:** 

https://deploy-preview-1276--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Go to the deploy preview and click the new navigation tab called **Map**. You should see that the sidebar is expanded by default.
2. Click the arrow in side menu bar to collapse and expand it. Scroll the placeholder area of the side bar and see that the title and nav arrow stay at the top.
3. Adjust the width to 600px or less or use a mobile device mode in dev tools. You should see that the sidebar is collapsed by default.
4. Click the nav arrow to expand it, scroll the placeholder area, and see that the title and arrow button stay at the top.

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectsMapView/components/MapDrawer.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  const toggleDrawer = () => {
+    setOpen(!open);
+  };
+
+  const handleTransitionEnd = () => {
+    if (open) {
+      setShowDrawerContent(true);
+    } else {
+      // Repaint canvas on drawer close to avoid blank space in map
+      ref.current && ref.current.resize();
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectsMapView/components/MapDrawer.js:119]: The map and the map drawer are in a flex container. The map container grows to fill the view as the drawer shrinks when it collapses. The canvas within the map container that displays the map tile layer does not so we need to trigger the map to resize when the side drawer collapse transition is over. You can see the issue by commenting out this line when running locally.

Still not happy with the flicker/shift when this fires (you can see it on the right side of the map when toggling in desktop mode) but I think is fine for now. I'll keep it in mind as this grows, and I'm open to any suggestions. 🙏

File: moped-editor/src/views/projects/projectsMapView/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  return (
+    <Page title="Projects Map">
+      <ErrorBoundary FallbackComponent={FallbackComponent}>
+        <Paper component="main" className={classes.content}>
+          <MapDrawer title={"Filter title"} ref={mapRef}>
+            <Stack spacing={1}>
+              {[...new Array(50)].map((i) => (
+                <Typography key={i}>Filter UI placeholder</Typography>
+              ))}
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectsMapView/index.js:37]: Temporary filler to test scrolling 🖱️

**Review by frankhereford — APPROVED** [review]
This looks so great Mike, thank you! What an awesome step in the direction of having an "all projects" map right at the top of the application. Beyond the fact it will be a whole new way to find relevant projects, it's going to broadcast that the app is well used and full of data. 🙌 🚢 🚢

**Review by roseeichelmann — APPROVED** [review]
this is looking great mike, so exciting to have this feature coming along!

**Review by roseeichelmann — APPROVED** [review]
this is looking great mike, so exciting to have this feature coming along!

**Comment by roseeichelmann** [comment]
idk why that left two reviews, i guess this gets double approvals from me :sunglasses:

File: moped-editor/src/views/projects/projectsMapView/components/MapDrawer.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    "& .MuiDrawer-paper": openedMixin(theme),
+  }),
+  ...(!open && {
+    ...containDrawerMixin(),
+    ...closedMixin(theme),
+    "& .MuiDrawer-paper": closedMixin(theme),
+  }),
+}));
+
+function DrawerContent({
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-editor/src/views/projects/projectsMapView/components/MapDrawer.js:59]: tbh, i'm just following patterns of code that I've been reading lately. Here are a couple of examples: [MUI drawer code](https://mui.com/material-ui/react-drawer/#responsive-drawer) and [Vercel's ecommerce starter template](https://github.com/vercel/commerce/blob/v1/site/components/search.tsx).

As far as React code goes, there used to be more importance in choosing which type of function declaration. [Here](https://medium.com/@rajwar67/a-guide-to-bind-this-keyword-in-react-js-256c7ee39970) is some history from when class components were common. After hooks were introduced and the convention switched to function components, there isn't a huge difference besides export syntax that I can think of at the moment.

If you want to go further down the JS rabbit hole 🕳️, [the MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) go into bindings and the `this` keyword.

tldr: Defaulting to arrow function is a-ok in modern React code!

File: moped-editor/src/views/projects/projectsMapView/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    height: "80vh",
+    display: "flex",
+  },
+}));
+
+const ProjectsMapView = () => {
+  const classes = useStyles();
+
+  /* Store map instance to call Mapbox GL methods where needed */
+  const mapRef = React.useRef();
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectsMapView/index.js:27]: Rose's comment about arrow fns reminded me to ask about how y'all feel about dot referencing React hooks this way? I think it is kind of a neat way to differentiate React hook vs. custom hooks. I've been seeing this lately in React projects too.
